internal/controller/nimcache_controller.go

```diff
@@ -495,73 +495,112 @@ func getSelectedProfiles(nimCache *appsv1alpha1.NIMCache) ([]string, error) {
-		pod := constructPodSpec(nimCache)
-		// Add nimCache as owner for watching on status change
-		if err := controllerutil.SetControllerReference(nimCache, pod, r.GetScheme()); err != nil {
-			return false, err
-		}
-		err := r.createPod(ctx, pod)
-		if err != nil {
-			logger.Error(err, "failed to create", "pod", pod.Name)
-			return false, err
-		}
+	// Model manifest is available only for NGC model pullers
+	if nimCache.Spec.Source.NGC == nil {
+		return false, nil
+	}
 
-		existingPod := &corev1.Pod{}
-		err = r.Get(ctx, client.ObjectKey{Name: pod.Name, Namespace: nimCache.Namespace}, existingPod)
-		if err != nil {
-			logger.Error(err, "failed to get pod for model selection", "pod", pod.Name)
-			return false, err
-		}
+	existingConfig := &corev1.ConfigMap{}
+	cmName := getManifestConfigName(nimCache)
+	err = r.Get(ctx, client.ObjectKey{Name: cmName, Namespace: nimCache.Namespace}, existingConfig)
+	if err != nil && client.IgnoreNotFound(err) != nil {
+		logger.Error(err, "failed to get configmap of the model manifest", "name", cmName)
+		return false, err
+	}
 
-		if existingPod.Status.Phase != corev1.PodRunning {
-			// requeue request with delay until the pod is ready
-			return true, nil
-		}
+	// No action if the configmap is already created
+	if err == nil {
+		return false, nil
+	}
 
-		// Extract manifest file
-		output, err := r.getPodLogs(ctx, existingPod)
-		if err != nil {
-			logger.Error(err, "failed to get pod logs for parsing model manifest file", "pod", pod.Name)
-			return false, err
-		}
+	// Create a configmap by extracting the model manifest
+	// Create a temporary pod for parsing model manifest
+	pod := constructPodSpec(nimCache)
+	// Add nimCache as owner for watching on status change
+	if err := controllerutil.SetControllerReference(nimCache, pod, r.GetScheme()); err != nil {
+		return false, err
+	}
+	err = r.createPod(ctx, pod)
+	if err != nil {
+		logger.Error(err, "failed to create", "pod", pod.Name)
+		return false, err
+	}
 
-		// Parse the file
-		manifest, err := nimparser.ParseModelManifestFromRawOutput([]byte(output))
-		if err != nil {
-			logger.Error(err, "Failed to parse model manifest from the pod")
-			return false, err
-		}
-		logger.V(2).Info("manifest file", "nimcache", nimCache.Name, "manifest", manifest)
+	existingPod := &corev1.Pod{}
+	err = r.Get(ctx, client.ObjectKey{Name: pod.Name, Namespace: nimCache.Namespace}, existingPod)
+	if err != nil {
+		logger.Error(err, "failed to get pod for model selection", "pod", pod.Name)
+		return false, err
+	}
 
-		// Create a ConfigMap with the model manifest file for re-use
-		err = r.createManifestConfigMap(ctx, nimCache, manifest)
-		if err != nil {
-			logger.Error(err, "Failed to create model manifest config map")
-			return false, err
-		}
+	if existingPod.Status.Phase != corev1.PodRunning {
+		// requeue request with delay until the pod is ready
+		return true, nil
+	}
 
+	// Extract manifest file
+	output, err := r.getPodLogs(ctx, existingPod)
```

Comment:
For model manifests -- do we have plan to add NGC api (or any other web apis) to query model metadata?

Reply:
pull an image and cat a file inside looks very hacky......

Reply:
that is not planned in the near term. The manifest is embedded in the NIM container itself. So we need to run the container to get that (either by `cp` or logging)